root/static/scripts/work/edit.js

```diff
@@ -115,15 +105,15 @@ const store = createStore<WorkForm, ActionT, (ActionT) => empty>(function (
 });
 
 function addLanguageToState(form: WorkForm): WorkForm {
-  return mutate<WritableWorkForm, _>(form, newForm => {
-    pushField(newForm.field.languages, null);
-  });
+  return mutate(form).update('field', 'languages', (fieldCtx) => {
+    pushField(fieldCtx, null);
+  }).final();
 }
 
 function removeLanguageFromState(form: WorkForm, i: number): WorkForm {
-  return mutate<WritableWorkForm, _>(form, newForm => {
-    newForm.field.languages.field.splice(i, 1);
-  });
+  return mutate(form).update('field', 'languages', 'field', (fieldCtx) => {
+    fieldCtx.write().splice(i);
```

Comment:
Is dropping `, 1` here (while it's left on other `splice()` calls) intentional?

Reply:
It was a bug and also the reason for the test change (which I thought changed because `last_index` was previously mutable). Good catch!